[BUG] attribute cannot be applied to types build error

When building SRT I run into the error 

> srt.h:623:69: error: 'deprecated' attribute cannot be applied to types
> SRT_API SRTSOCKET srt_socket       (int af, int type, int protocol) SRT_ATR_DEPRECATED;

> srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
> #define SRT_ATR_DEPRECATED [[deprecated]]

To Reproduce
Steps to reproduce the behavior:
Download and unpack:
[srtbuildtest.zip](https://github.com/Haivision/srt/files/4116293/srtbuildtest.zip)
Then ->
cmake .
make

(you might have to lower the cmake version I'm using cmake 3.15)

**Expected behavior**
```
/Users/anderscedronius/git/srtbuildtest/srt/srtcore/srt.h:623:69: error: 'deprecated' attribute cannot be applied to types
SRT_API SRTSOCKET srt_socket       (int af, int type, int protocol) SRT_ATR_DEPRECATED;
                                                                    ^
/Users/anderscedronius/git/srtbuildtest/srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]
                             ^
/Users/anderscedronius/git/srtbuildtest/srt/srtcore/srt.h:628:72: error: 'deprecated' attribute cannot be applied to types
static inline int srt_bind_peerof(SRTSOCKET u, UDPSOCKET sys_udp_sock) SRT_ATR_DEPRECATED;
                                                                       ^
/Users/anderscedronius/git/srtbuildtest/srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]
```

**Desktop (please provide the following information):**
 - OS: MacOS
 - SRT Version / commit ID: c59d413

**Additional context**
my solution (don’t know if its viable)  was to move the SRT_ATR_DEPRECATED  to the start of the line (623, 628) like this ->
![image](https://user-images.githubusercontent.com/14168229/73162469-25df3280-40ee-11ea-9b12-6cd7b7b72b8d.png)
Then everything builds.

No problem.. Here we go with verbose information >

CMake

```
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /usr/local/lib/libcrypto.dylib (found version "1.0.2t")  
-- Configuring done
-- Generating done
```

```
make
/usr/local/Cellar/cmake/3.16.2/bin/cmake -S/Users/anderscedronius/Downloads/srtbuildtest -B/Users/anderscedronius/Downloads/srtbuildtest --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E cmake_progress_start /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/project_srt.dir/build.make CMakeFiles/project_srt.dir/depend
cd /Users/anderscedronius/Downloads/srtbuildtest && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E cmake_depends "Unix Makefiles" /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles/project_srt.dir/DependInfo.cmake --color=
Scanning dependencies of target project_srt
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/project_srt.dir/build.make CMakeFiles/project_srt.dir/build
[ 10%] Creating directories for 'project_srt'
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/srt
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/srt
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/tmp
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-mkdir
[ 20%] Performing download step (git clone) for 'project_srt'
/usr/local/Cellar/cmake/3.16.2/bin/cmake -P /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/tmp/project_srt-gitclone.cmake
Cloning into 'srt'...
remote: Enumerating objects: 4, done.        
remote: Counting objects: 100% (4/4), done.        
remote: Compressing objects: 100% (4/4), done.        
remote: Total 6901 (delta 0), reused 3 (delta 0), pack-reused 6897        
Receiving objects: 100% (6901/6901), 7.76 MiB | 5.53 MiB/s, done.
Resolving deltas: 100% (5330/5330), done.
Already on 'master'
Your branch is up to date with 'origin/master'.
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-download
[ 30%] No patch step for 'project_srt'
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E echo_append
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-patch
[ 40%] No update step for 'project_srt'
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E echo_append
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-update
[ 50%] Performing configure step for 'project_srt'
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /Users/anderscedronius/Downloads/srtbuildtest/srt/configure
Running: cmake /Users/anderscedronius/Downloads/srtbuildtest/srt -DOPENSSL_INCLUDE_DIR=/usr/local/opt/openssl/include -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib/libcrypto.a
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- BUILD TYPE: Release
-- LOGGING: ENABLED
-- USE_BUSY_WAITING: OFF (default)
-- No WITH_COMPILER_PREFIX - using C++ compiler /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Checking for module 'openssl libcrypto'
--   No package 'openssl' found
No package 'libcrypto' found
-- Found OpenSSL: /usr/local/lib/libcrypto.dylib (found version "1.0.2t")  
-- SSL via find_package(OpenSSL): -I /usr/local/opt/openssl/include -l;/usr/local/lib/libssl.dylib;/usr/local/lib/libcrypto.dylib
-- ENCRYPTION: ENABLED, using: openssl libcrypto
-- SSL libraries: /usr/local/lib/libssl.dylib;/usr/local/lib/libcrypto.dylib
-- COMPILER: Clang (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++) - GNU compat
-- DETECTED SYSTEM: DARWIN;  OSX=1
-- ADDING TRANSITIVE LINK DEP to:srt_shared :  
-- ADDING TRANSITIVE LINK DEP to:srt_static :  
-- INSTALL DIRS: bin=bin lib=lib shlib=lib include=include
-- C++ VERSION: Setting C++11 compat flag for gnu compiler
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/anderscedronius/Downloads/srtbuildtest/srt
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-configure
[ 60%] Performing build step for 'project_srt'
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && make srt_static
Scanning dependencies of target srt_virtual
[  3%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/api.cpp.o
[  6%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/buffer.cpp.o
[  9%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/cache.cpp.o
[ 12%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/channel.cpp.o
[ 15%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/common.cpp.o
[ 18%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/core.cpp.o
[ 21%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/crypto.cpp.o
[ 25%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/epoll.cpp.o
[ 28%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/fec.cpp.o
/Users/anderscedronius/Downloads/srtbuildtest/srt/srtcore/fec.cpp:976:20: warning: unused function 'CellMark' [-Wunused-function]
static inline char CellMark(const std::deque<bool>& cells, int index)
                   ^
1 warning generated.
[ 31%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/handshake.cpp.o
[ 34%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/list.cpp.o
[ 37%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/md5.cpp.o
[ 40%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/packet.cpp.o
[ 43%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/packetfilter.cpp.o
[ 46%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/queue.cpp.o
[ 50%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/congctl.cpp.o
[ 53%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/srt_c_api.cpp.o
[ 56%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/window.cpp.o
[ 59%] Building C object CMakeFiles/srt_virtual.dir/srtcore/srt_compat.c.o
[ 62%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/sync.cpp.o
[ 65%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/cryspr.c.o
[ 68%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/cryspr-openssl.c.o
[ 71%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt.c.o
[ 75%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_ctx_rx.c.o
[ 78%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_ctx_tx.c.o
[ 81%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_rx.c.o
[ 84%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_sa.c.o
[ 87%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_tx.c.o
[ 90%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_xpt_srt.c.o
[ 93%] Building CXX object CMakeFiles/srt_virtual.dir/haicrypt/haicrypt_log.cpp.o
[ 93%] Built target srt_virtual
Scanning dependencies of target srt_static
[ 96%] Building C object CMakeFiles/srt_static.dir/cmake_object_lib_support.c.o
[100%] Linking CXX static library libsrt.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsrt.a(haicrypt_log.cpp.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libsrt.a(haicrypt_log.cpp.o) has no symbols
[100%] Built target srt_static
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-build
[ 70%] No install step for 'project_srt'
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E echo_append
cd /Users/anderscedronius/Downloads/srtbuildtest/srt && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-install
[ 80%] Completed 'project_srt'
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E make_directory /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles/project_srt-complete
/usr/local/Cellar/cmake/3.16.2/bin/cmake -E touch /Users/anderscedronius/Downloads/srtbuildtest/project_srt-prefix/src/project_srt-stamp/project_srt-done
[ 80%] Built target project_srt
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/srtbuildtest.dir/build.make CMakeFiles/srtbuildtest.dir/depend
cd /Users/anderscedronius/Downloads/srtbuildtest && /usr/local/Cellar/cmake/3.16.2/bin/cmake -E cmake_depends "Unix Makefiles" /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest /Users/anderscedronius/Downloads/srtbuildtest/CMakeFiles/srtbuildtest.dir/DependInfo.cmake --color=
Scanning dependencies of target srtbuildtest
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/srtbuildtest.dir/build.make CMakeFiles/srtbuildtest.dir/build
[ 90%] Building CXX object CMakeFiles/srtbuildtest.dir/main.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -I/Users/anderscedronius/Downloads/srtbuildtest/srt  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk   -std=gnu++1z -o CMakeFiles/srtbuildtest.dir/main.cpp.o -c /Users/anderscedronius/Downloads/srtbuildtest/main.cpp
In file included from /Users/anderscedronius/Downloads/srtbuildtest/main.cpp:2:
/Users/anderscedronius/Downloads/srtbuildtest/srt/srtcore/srt.h:623:69: error: 'deprecated' attribute cannot be applied to types
SRT_API SRTSOCKET srt_socket       (int af, int type, int protocol) SRT_ATR_DEPRECATED;
                                                                    ^
/Users/anderscedronius/Downloads/srtbuildtest/srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]
                             ^
/Users/anderscedronius/Downloads/srtbuildtest/srt/srtcore/srt.h:628:72: error: 'deprecated' attribute cannot be applied to types
static inline int srt_bind_peerof(SRTSOCKET u, UDPSOCKET sys_udp_sock) SRT_ATR_DEPRECATED;
                                                                       ^
/Users/anderscedronius/Downloads/srtbuildtest/srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]
                             ^
2 errors generated.
make[2]: *** [CMakeFiles/srtbuildtest.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/srtbuildtest.dir/all] Error 2
make: *** [all] Error 2
```

--- a/srtcore/srt.h
+++ b/srtcore/srt.h
@@ -74,23 +74,44 @@ written by
 // for all lesser standard use compiler-specific attributes).
 #if defined(SRT_NO_DEPRECATED)
 
+// Unused: do not issue an unused variable warning
 #define SRT_ATR_UNUSED
+
+// Deprecated: function or symbol is deprecated
+// The *_PX version is the prefix attribute, which applies only
+// to functions (Microsoft compilers).
+
+// When deprecating a function, mark it:
+//
+// SRT_ATR_DEPRECATED_PX retval function(arguments) SRT_ATR_DEPRECATED;
+//
 #define SRT_ATR_DEPRECATED
+#define SRT_ATR_DEPRECATED_PX
+
+// Nodiscard: issue a warning if the return value was discarded.
 #define SRT_ATR_NODISCARD
 
 #elif defined(__cplusplus) && __cplusplus > 201406
 
 #define SRT_ATR_UNUSED [[maybe_unused]]
-#define SRT_ATR_DEPRECATED [[deprecated]]
+#define SRT_ATR_DEPRECATED
+#define SRT_ATR_DEPRECATED_PX [[deprecated]]
 #define SRT_ATR_NODISCARD [[nodiscard]]
 
 // GNUG is GNU C/C++; this syntax is also supported by Clang
-#elif defined( __GNUC__)
+#elif defined(__GNUC__)
 #define SRT_ATR_UNUSED __attribute__((unused))
+#define SRT_ATR_DEPRECATED_PX
 #define SRT_ATR_DEPRECATED __attribute__((deprecated))
 #define SRT_ATR_NODISCARD __attribute__((warn_unused_result))
+#elif defined(_MSC_VER)
+#define SRT_ATR_UNUSED __pragma(warning(suppress: 4100 4101))
+#define SRT_ATR_DEPRECATED_PX __declspec(deprecated)
+#define SRT_ATR_DEPRECATED // no postfix-type modifier
+#define SRT_ATR_NODISCARD _Check_return_
 #else
 #define SRT_ATR_UNUSED
+#define SRT_ATR_DEPRECATED_PX
 #define SRT_ATR_DEPRECATED
 #define SRT_ATR_NODISCARD
 #endif
@@ -206,7 +227,7 @@ typedef enum SRT_SOCKOPT {
 #define SRT_DEPRECATED_OPTION(value) ((SRT_SOCKOPT [[deprecated]])value)
 #else
     // Older (pre-C++11) compilers use gcc deprecated applied to a typedef
-    typedef SRT_ATR_DEPRECATED SRT_SOCKOPT SRT_SOCKOPT_DEPRECATED;
+    typedef SRT_ATR_DEPRECATED_PX SRT_SOCKOPT SRT_SOCKOPT_DEPRECATED SRT_ATR_DEPRECATED;
 #define SRT_DEPRECATED_OPTION(value) ((SRT_SOCKOPT_DEPRECATED)value)
 #endif
 
@@ -444,8 +465,8 @@ enum CodeMinor
     MN_CONGESTION      =  4
 };
 
-static const enum CodeMinor MN_ISSTREAM SRT_ATR_DEPRECATED = (enum CodeMinor)(9);
-static const enum CodeMinor MN_ISDGRAM SRT_ATR_DEPRECATED = (enum CodeMinor)(10);
+SRT_ATR_DEPRECATED_PX static const enum CodeMinor MN_ISSTREAM SRT_ATR_DEPRECATED = (enum CodeMinor)(9);
+SRT_ATR_DEPRECATED_PX static const enum CodeMinor MN_ISDGRAM SRT_ATR_DEPRECATED = (enum CodeMinor)(10);
 
 // Stupid, but effective. This will be #undefined, so don't worry.
 #define MJ(major) (1000 * MJ_##major)
@@ -620,12 +641,12 @@ SRT_API       int srt_cleanup(void);
 //
 // DEPRECATED: srt_socket with 3 arguments. All these arguments are ignored
 // and socket creation doesn't need any arguments. Use srt_create_socket().
-SRT_API SRTSOCKET srt_socket       (int af, int type, int protocol) SRT_ATR_DEPRECATED;
-SRT_API SRTSOCKET srt_create_socket();
+SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
+SRT_API       SRTSOCKET srt_create_socket();
 SRT_API       int srt_bind         (SRTSOCKET u, const struct sockaddr* name, int namelen);
 SRT_API       int srt_bind_acquire (SRTSOCKET u, UDPSOCKET sys_udp_sock);
 // Old name of srt_bind_acquire(), please don't use
-static inline int srt_bind_peerof(SRTSOCKET u, UDPSOCKET sys_udp_sock) SRT_ATR_DEPRECATED;
+SRT_ATR_DEPRECATED_PX static inline int srt_bind_peerof(SRTSOCKET u, UDPSOCKET sys_udp_sock) SRT_ATR_DEPRECATED;
 static inline int srt_bind_peerof  (SRTSOCKET u, UDPSOCKET sys_udp_sock) { return srt_bind_acquire(u, sys_udp_sock); }
 SRT_API       int srt_listen       (SRTSOCKET u, int backlog);
 SRT_API SRTSOCKET srt_accept       (SRTSOCKET u, struct sockaddr* addr, int* addrlen);

